### ISSUE TITLE
Bundled macOS .app has a yellow title bar in 10.14 (Mojave)

### ISSUE THREAD

**Issue by tazzben** [pyinstaller/pyinstaller#3770]
Tkinter based scripts that are bundled with PyInstaller with Python 3.6.5 or higher (including 3.7) result in a  window title bar that is yellow (where it is the standard look when the script is run directly).  This is using the version of python posted at python.org.  I've posted a screenshot at the end of this post for you to see what I'm talking about.

It is of note, that versions of python prior to 3.6.5 did not bundle their own version of TK.  Versions using the system TK run, but show up as blank in Mojave (a button, for instance, can be pressed -- and works --, but the text is white or otherwise not visible).  The same bundle works on earlier versions of macOS as expected.

I believe this is related to some of the other issues currently noted about tkinter on macOS with PyInstaller.  As a baseline, I've been using the following sample script, which is posted at python.org:

```
import tkinter as tk

class Application(tk.Frame):
    def __init__(self, master=None):
        super().__init__(master)
        self.pack()
        self.create_widgets()

    def create_widgets(self):
        self.hi_there = tk.Button(self)
        self.hi_there["text"] = "Hello World\n(click me)"
        self.hi_there["command"] = self.say_hi
        self.hi_there.pack(side="top")

        self.quit = tk.Button(self, text="QUIT", fg="red",
                              command=root.destroy)
        self.quit.pack(side="bottom")

    def say_hi(self):
        print("hi there, everyone!")

root = tk.Tk()
app = Application(master=root)
app.mainloop()
```

If this script is bundled single directory style, the .app file does not start initially.   However, it will start if you manually create the folders "tk" and "tcl" in the MacOS folder within the .app bundle.  For the .app file to work on a computer without python 3.7 preinstalled, you will also have to copy the tk8.6 and tcl8.6 folders, icons, and config files (*.sh) from /Library/Frameworks/Python.framework/Versions/3.7/lib/ and paste them in the MacOS folder in the bundle.  You should *not* copy the system TCL/TK files (as suggested in other issues) as they will conflict -- I imagine that is a fix in versions prior to 3.6.5.     

This works in every other respect, except that the title bar is yellow.  As the title bar is only yellow after being bundled with PyInstaller, I have to believe it has something to do with the bundling process, but I'm not sure what the issue is.

<img width="84" alt="screen shot 2018-09-27 at 1 11 22 pm" src="https://user-images.githubusercontent.com/303181/46165695-e6cd7280-c256-11e8-8ec2-096bbf0ff077.png">

**Comment by nelsonjchen** [comment]
Does dark mode have or not have an effect?

**Comment by tazzben** [comment]
The  bar is yellow in both dark and light mode.

**Comment by AmbroiseGrau** [comment]
Hello, 
Do you have by any chance another installation of Tcl/Tk (such as coming from ActiveTcl)? On one of the computer I'm using, I have in my folder `/Library/Frameworks/Tcl.framework/Versions/` a '8.6' folder (as well as a `Current` alias linking to this one), whereas on another computer the Version folder is empty (similarly for Tk).
In the former case, I also get these yellow title bar, whereas in the latter it has the normal colour. Similarly, when I gave it to run to a friend who doesn't have anything specially installed (i.e. no Python or Tcl installation apart from the defaults present in MacOS so `/Library/Frameworks/Tcl.framework` doesn't even exists), it was also the normal grey colour. Thus I think that this does not comes from the way PyInstaller bundle the application (since also bundling with py2app gives the same interesting change of colours), but more from the installation that exist on your computer (as it might use the version 8.5 present in `/System/Library/Tcl.framework` when it doesn't have one in `/Library/Framework/Tcl.framework`).

I didn't know your way of adding this two folders in order to make it work, so I came up with another fix modifying the source file, directly including the libraries (I will post something about that in the thread #3753 that is more directly related to it).

It seems that the libraries added from Python are not used as they are, but that the version present on `/Library/Framework/Tcl.framework` is still predominant. I don't understand exactly why; it might be the order in which it will look for the path of Tcl and Tk (as running directly the code via Python interpreter gives the normal grey colour as you pointed out).

**Comment by codebykevin** [comment]
The situation here is that the changes in macOS drawing API's have caused significant issues with Tk on the Mac; we are working on an update that will be released with Tk 8.6.9, due soon. Updating Tk to the new version will be the best way to ensure avoiding issues with Tk on Mojave.

**Comment by htgoebel** [comment]
I take this comment as: This is not y PyIsntaller issue.
